java/statistical-query-group-and-sort/README.md

```diff
@@ -1,26 +1,50 @@
 
-![Statistical Query Group and Sort App](statistical-query-group-and-sort.png)
+Query a feature table for statistics, grouping and sorting by different fields.
+
+![Image of statistical query group and sort](statistical-query-group-and-sort.png)
+
+## Use case
+
+You can use statistical queries, grouping and sorting to process large amounts of data saved in feature tables. This is helpful for identifying trends and relationships within the data, which can be used to support further interpretations and decisions. For example, a health agency can use information on medical conditions occurring throughout a country to identify at-risk areas or demographics, and decide on further action and preventive measures.
 
 ## How to use the sample
-1. Select a combination of fields and statistic types to include in the query. Use the 'Add' button to add selections to the list.
-1. (Optional) Choose one or more fields to group the results on by checking boxes next to field names in the 'Group Field(s)' list. Grouping by SUB_REGION, for example, will show results grouped (summarized) for each region ('Pacific', 'Mountain', etc).
-1. (Optional) Choose one or more fields to order results by. Use the '>>' and '<<' to add or remove fields to sort by. NOTE: Only those fields selected for grouping are valid choices for ordering results.
-1. Execute the query by hitting the 'Get Statistics' button. Results will be displayed in an expandable list view in a new activity. Results will be grouped and sorted according to the chosen fields.
+
+The sample will start with some default options selected. You can immediately click the "Get Statistics" button to see the results for these options. There are several ways to customize your queries:
+
+* You can add statistic definitions to the top-left table using the drop down menus and "Add" button. Select a table row and click "Remove statistic" to remove a definition.
+
+* To change the Group-by fields, check the box by the field you want to group by in the bottom-left list view. 
+
+* To change the Order-by fields, select a Group-by field (it must be checked) and click the ">>" button to add it to the Order-by table. To remove a field from the Order-by table, select it and click the "<<" button. To change the sort order of the Order-by field, the cells of the "Sort Order" column are combo-boxes that may be either ASCENDING or DESCENDING.
 
 ## How it works
-1. Create `StatisticsQueryParameters` with a list of `StatisticDefinition`s.
-1. Add a list of fields (as strings) to group by with `.getGroupByFieldNames().add(...)`.
-1. Add a list of fields (as strings) to order by with `getOrderByFields().add(...)`.
-1. Get the statistics query result from the `ServiceFeatureTable` with `.queryStatisticsAsync(...)` on the `StatisticsQueryParameters`.
+
+1. Create a `ServiceFeatureTable` using the URL of a feature service and load the table.
+2. Get the feature tables field names list with `featureTable.getFields()`.
```

Comment:
Might be worth changing this a bit to show how the sample loops through the fields:

```
      // fill array with field names
      mFieldNameList = new ArrayList<>();
      for (int i = 0; i < mUsStatesFeatureTable.getFields().size(); i++) {
        String fieldName = mUsStatesFeatureTable.getFields().get(i).getName();
        mFieldNameList.add(fieldName);
      }
```

Reply:
we always keep the readme high level: so this level of detail would be too much. the user can see the nitty gritty of how it fits together in the code but can refer to the readme to see an overview of the important methods/classes.